packages/rest/src/parse-json.ts

```diff
@@ -13,14 +13,36 @@
 // replacement for `JSON.parse` but we need to instruct `body-parser` to honor
 // a `reviver` function.
 
+const isMatched = (key: string, pattern: string) =>
+  pattern === key || key.indexOf(`${pattern}.`) === 0;
+
 /**
  * Factory to create a reviver function for `JSON.parse` to sanitize keys
  * @param reviver - Reviver function
+ * @param prohibitedKeys - An array of keys to be rejected
  */
-export function sanitizeJsonParse(reviver?: (key: any, value: any) => any) {
+export function sanitizeJsonParse(
+  reviver?: (key: any, value: any) => any,
+  prohibitedKeys?: string[],
+) {
   return (key: string, value: any) => {
-    if (key === '__proto__')
-      throw new Error('JSON string cannot contain "__proto__" key.');
+    if (key === '__proto__') {
+      // Reject `__proto__`
+      throw new Error(`JSON string cannot contain "${key}" key.`);
```

Comment:
is it okay throws an Error? i think this returns a 500 error and could stop the application may be should throw a 422 (Unprocessable Entity)

Reply:
See https://github.com/strongloop/loopback-next/blob/ae0b9936e7eadbf6f0ee7c72e1a04b87dda7c2c5/examples/todo/src/__tests__/acceptance/todo.acceptance.ts#L183. JSON parsing errors are wrapped into `400`.